server/backend/neighbors.py

```diff
@@ -0,0 +1,227 @@
+import re
+from itertools import groupby
+from pathlib import Path
+
+import pandas as pd
+import plydata as ply
+from gensim.models import KeyedVectors, Word2Vec
+from joblib import Memory, Parallel, delayed
+
+from .config import (
+    USE_MEMMAP, MATERIALIZE_MODELS,
+    PARALLEL_POOLS, PARALLELIZE_QUERY, PARALLEL_BACKEND
+)
+from .tracking import ExecTimer
+
+# the root for the word-lapse-models datafiles
+data_folder = Path("./data")
+
+# stores cached word model references; used if MATERIALIZE_MODELS is true
+# (pre-populated on server startup if .config.WARM_CACHE is true)
+word_models = None
+
+
+# ========================================================================
+# === extract_frequencies(), cutoff_points()
+# ========================================================================
+
+def extract_frequencies(tok: str):
+    # Extract the frequencies
+    frequency_table = pd.read_csv(
+        data_folder / Path("all_tok_frequencies.tsv.xz"), sep="\t"
+    )
+    frequency_table >> ply.slice_rows(5)
```

Comment:
Could remove this and all the other lines similar to this one. Don't need to output dataframe rows to standard out.